packages/navie/src/services/context-service.ts

```diff
@@ -35,8 +35,9 @@ export default class ContextService {
         searchTerms.push(...additionalTerms);
       }
 
-      const filters = { ...options.buildContextFilters() };
-
+      // Locations are not considered when searching for context. Use the `locationContext` method
+      // to retrieve context for specific files.
+      const filters = { ...options.buildContextFilters(), locations: undefined };
```

Comment:
This reinforces my intuition that this really wants to be one method from the caller's perspective.